src/depth_packet_processor.cpp

```diff
@@ -98,4 +100,64 @@ void DepthPacketProcessor::setFrameListener(libfreenect2::FrameListener *listene
   listener_ = listener;
 }
 
+  DumpDepthPacketProcessor::DumpDepthPacketProcessor() : depth_frame(new Frame(512, 424, 4)), ir_frame(new Frame(512, 424,4)) {
+}
+
+DumpDepthPacketProcessor::~DumpDepthPacketProcessor(){
+  delete depth_frame;
+  delete ir_frame;
+  delete[] p0table_;
+  delete[] xtable_;
+  delete[] ztable_;
+  delete[] lut_;
+}
+
+void DumpDepthPacketProcessor::process(const DepthPacket &packet) {
+    ir_frame->height = 1;
+    ir_frame->width = 1;
+    ir_frame->bytes_per_pixel = packet.buffer_length;
+    ir_frame->timestamp = packet.timestamp;
+    ir_frame->sequence = packet.sequence;
+    ir_frame->data = packet.buffer;
```

Comment:
done.  (thanks for the extensive comment/sample)

I reversed the order of the deletes since the depth_frame owns the data pointer, so we don't want to delete it before handing `ir_frame` off to the depth handler.